docs/contributor/adr/0007-rate-limit.md

```diff
@@ -0,0 +1,30 @@
+# Rate Limit
+
+## Status
+
+Draft
+
+## Context
+
+It has been decided to indroduce rate limit functionality into Kyma platform to allow users to set rate limit functionality on the service mesh layer, therefore allowing to consume intended service mesh functionality which is abstracting away networking concerns outside applications inside the mesh.
+Since Istio is an underlying service mesh responsible for the workloads networking across Kyma clusters, therefore it is the Istio which allows to configure such functionality.
+At this moment it is not possible to introduce global rate limit provided by Istio service mesh because of the technical or legal limitations, related to the deployment of key-value stores in the cluster.
+Major difference between local and global rate limit is that local rate limiting is applied directly at the sidecar proxy level, controlling the rate of requests on a per-instance basis. Global rate limiting is applied accross multiple instances of a service or across, multiple services, providing a coorinated rate limit within the mesh.
+In case of introduction the global rate limit functionality in the future, new ADR will be created.
+
+## Decision
+
+It was decided to implement local rate limit as the only rate limiting option at this time.
+New CRD in the Istio module will be introduced to allow its configuration.
+In the future, global rate limit might be introduced. The decision has not been made yet, therefore it has been decided to name the new CRD `RateLimit` to not to indicate, that global rate limit will be provided. By naming current CRD for example `LocalRateLimit` logical would be to expect also `GlobalRateLimit` to be existent at some point.
+Currently the possibilites for the rate limit functionality are limited to restrict number of requests, without any other criteria like header presence or header value. 
+CRD is structured in a way that allows further extensions, allowing to configure different rate limit criteria, or to introduce global rate limit.
+
+## Consequences
```

Comment:
Maybe we should add a consequence that rate limited requests will result in a status code 429?